### PR TITLE
fix: upgrade and adjust the tomo connect to pave way for broadcast BB…

### PR DESCRIPTION
**Pull request by jrwbabylonlab**
Registry require the proto ts to be upgraded to produce ts types instead of methods.
Alternately, we can generate in this repo for now. but i will leave it for a seperate PR

https://github.com/babylonlabs-io/babylon-proto-ts/issues/14

### CODE REVIEW

**Review by totraev — APPROVED** [review]

File: src/app/context/wallet/CosmosWalletProvider.tsx [hunk 1]
```diff
@@ -10,56 +11,53 @@ import {
...
   }, []);
 
   const connectCosmos = useCallback(async () => {
     if (!providers.cosmosProvider) return;
 
     try {
-      const chainID = providers.cosmosProvider.getChainId();
-      const cosmosInfo =
-        await providers.cosmosProvider.provider.getKey(chainID);
+      await providers.cosmosProvider.connectWallet();
```
Comment (totraev) [inline src/app/context/wallet/CosmosWalletProvider.tsx:54]: @jrwbabylonlab why we need this method here?

**Review by totraev — APPROVED** [review]

File: src/app/context/wallet/CosmosWalletProvider.tsx [hunk 2]
```diff
@@ -1,3 +1,4 @@
+import { SigningStargateClient } from "@cosmjs/stargate";
```
Comment (gbarkhatov) [inline src/app/context/wallet/CosmosWalletProvider.tsx:1]: We are importing this, but it is not a direct dependency of the app (we don't have `@cosmjs/stargate` in the package.json)

File: src/utils/wallet/bbnRegistry.ts [hunk 3]
```diff
@@ -0,0 +1,6 @@
+import { Registry } from "@cosmjs/proto-signing";
```
Comment (gbarkhatov) [inline src/utils/wallet/bbnRegistry.ts:1]: Same here, not a direct dependency

File: src/app/context/wallet/CosmosWalletProvider.tsx [hunk 4]
```diff
@@ -10,56 +11,53 @@ import {
...
-  pubKey: string;
   connected: boolean;
   disconnect: () => void;
   open: () => void;
-  signArbitrary: (message: string) => Promise<any>;
+  getSigningStargateClient(): Promise<SigningStargateClient>;
 }
 
+const getSigningStargateClientDefault = async () => {
+  throw new Error("Not implemented");
```
Comment (gbarkhatov) [inline src/app/context/wallet/CosmosWalletProvider.tsx:27]: Should we add a todo here?

**Review by gbarkhatov — APPROVED** [review]

**Pull request merged**